### PR TITLE
CVS-46632 Parse stateful specific inputs in http handler

### PR DESCRIPTION
**Pull request by mzegla**

### CODE REVIEW

File: src/rest_parser.cpp [hunk 1]
```diff
@@ -49,6 +49,41 @@ void RestParser::removeUnusedInputs() {
...
+            proto.add_uint32_val(value.GetUint());
+        else
+            return false;
+    }
+    return true;
+}
+
+bool RestParser::parseSpecialInput(rapidjson::Value& doc, tensorflow::TensorProto& proto, const std::string& tensorName) {
+    // Special tensors are given in 1 dimentional array
+    if (doc.GetArray()[0].IsArray())
```
Comment (dkalinowski) [inline src/rest_parser.cpp:76]: Should we ensure that GetArray returns an array with at least one element?

File: src/statefulmodelinstance.cpp [hunk 2]
```diff
@@ -27,15 +27,41 @@ using namespace InferenceEngine;
...
+
     if (proto.uint64_val_size() == 1) {
         sequenceId = proto.uint64_val(0);
         return StatusCode::OK;
     }
     return StatusCode::SEQUENCE_ID_BAD_TYPE;
 }
 
-const Status extractSequenceControlInput(const tensorflow::TensorProto& proto, uint32_t& sequenceControlInput) {
+const Status StatefulModelInstance::extractSequenceControlInput(const tensorflow::TensorProto& proto, uint32_t& sequenceControlInput) {
```
Comment (dkalinowski) [inline src/statefulmodelinstance.cpp:51]: These methods: `extractSequenceControlInput` and `extractSequenceId` are very similar and can be used for different purposes than handling control input/sequence id too. If the only difference is type (uint32 vs uint64) then maybe name them accordingly?

Comment (dkalinowski) [reply src/statefulmodelinstance.cpp:51]: Like `extractOneUint32` and `extractOneUint64`?

File: src/rest_utils.cpp [hunk 3]
```diff
@@ -48,59 +57,107 @@ Status makeJsonFromPredictResponse(
...
+                    return status;
+            } else {
+                for (size_t i = 0; i < tensor.tensor_content().size(); i += sizeof(uint8_t))
+                    tensor.add_int_val(*reinterpret_cast<uint8_t*>(tensor.mutable_tensor_content()->data() + i));
             }
             break;
         case DataType::DT_INT64:
-            for (size_t i = 0; i < tensor.tensor_content().size(); i += sizeof(int64_t)) {
-                tensor.add_int64_val(*reinterpret_cast<int64_t*>(tensor.mutable_tensor_content()->data() + i));
+            if (seekDataInValField) {
```
Comment (dkalinowski) [inline src/rest_utils.cpp:134]: Please add additional unit tests for `makeJsonFromPredictResponse`

Comment (dkalinowski) [reply src/rest_utils.cpp:134]: `rest_utils_test.cpp`

**Review by rasapala — APPROVED** [review]
LGTM

File: src/statefulmodelinstance.cpp (hunk 2 above)
Comment (mzegla) [reply src/statefulmodelinstance.cpp:51]: I don't see any other usage of these methods than in sequence_id and sequence_control_input extraction in the near future. 
Can we stay with those very specific names to make the purpose of these methods obvious for now and when any other use case for these methods appears we would rename them?

Comment (dkalinowski) [reply src/statefulmodelinstance.cpp:51]: Seems fair

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**